### PR TITLE
Added slack settings OSSK-382

### PR DESCRIPTION
**Pull request by htahir1**
## Describe changes
Closes OSSK-382

## Pre-requisites
Please ensure you have done the following:
- [ ] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] I have based my new branch on `develop` and the open PR is targeting `develop`. If your branch wasn't based on develop read [Contribution guide on rebasing branch to develop](https://github.com/zenml-io/zenml/blob/main/CONTRIBUTING.md#-pull-requests-rebase-your-branch-on-develop).
- [ ] If my changes require changes to the dashboard, these changes are communicated/requested.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

File: src/zenml/integrations/slack/flavors/slack_alerter_flavor.py [hunk 1]
```diff
@@ -26,18 +27,24 @@
...
+    """
+
+    default_slack_channel_id: Optional[str] = None
+
+
+class SlackAlerterConfig(BaseAlerterConfig, SlackAlerterSettings):
     """Slack alerter config.
 
-    Attributes:
+    Args:
```
Comment (htahir1) [inline src/zenml/integrations/slack/flavors/slack_alerter_flavor.py:43]:
```suggestion
    Attributes:
```

File: src/zenml/integrations/slack/flavors/slack_alerter_flavor.py [hunk 2]
```diff
@@ -26,18 +27,24 @@
     from zenml.integrations.slack.alerters import SlackAlerter
 
 
-class SlackAlerterConfig(BaseAlerterConfig):
+class SlackAlerterSettings(BaseSettings):
+    """Settings for the Slack alerter.
+
+    Args:
```
Comment (htahir1) [inline src/zenml/integrations/slack/flavors/slack_alerter_flavor.py:33]:
```suggestion
    Attributes:
```

**Comment by htahir1** [comment]
@coderabbit review

**Review by avishniakov — APPROVED** [review]
LGTM, assuming tests will pass. I triggered Slow CI as well.

File: src/zenml/integrations/slack/alerters/slack_alerter.py [hunk 3]
```diff
@@ -72,6 +74,15 @@ def config(self) -> SlackAlerterConfig:
         """
         return cast(SlackAlerterConfig, self._config)
 
+    @property
+    def settings_class(self) -> Type[SlackAlerterSettings]:
+        """Settings class for the Kubeflow orchestrator.
```
Comment (htahir1) [inline src/zenml/integrations/slack/alerters/slack_alerter.py:79]:
```suggestion
        """Settings class for the Slack alerter.
```

**Comment by htahir1** [comment]
@coderabbitai review

**Review by strickvl — APPROVED** [review]

**Comment by htahir1** [comment]
@avishniakov @strickvl I had to revert the multi-inheritenance of the Config because of some linting issues, so I used `slack_channel_id` in settings and `default_slack_channel_id` in the config. I know its a bit weird but I think it somehow makes sense in this case. Let me know if you disagree

File: src/zenml/integrations/slack/flavors/slack_alerter_flavor.py [hunk 4]
```diff
@@ -31,20 +31,22 @@ class SlackAlerterSettings(BaseSettings):
...
+    slack_channel_id: Optional[str] = None
 
 
-class SlackAlerterConfig(BaseAlerterConfig, SlackAlerterSettings):
+class SlackAlerterConfig(BaseAlerterConfig):
     """Slack alerter config.
 
     Attributes:
         slack_token: The Slack token tied to the Slack account to be used.
+        default_slack_channel_id: The default ID of the Slack channel to use for communication.
```
Comment (strickvl) [inline src/zenml/integrations/slack/flavors/slack_alerter_flavor.py:45]:
```suggestion
        default_slack_channel_id: The ID of the default Slack channel to use for communication.
```

Maybe?

**Review by avishniakov — APPROVED** [review]

**Pull request merged**